src/drv/stm32cube/bsp_i2c_master.c

```diff
@@ -286,8 +325,7 @@ bsp_status_t bsp_i2c_master_read_u8(bsp_dev_i2c_t dev_num, uint8_t* rx_data)
 		set_sda_float();
 		i2c_sw_delay();
 
-		set_scl_float();
-		i2c_sw_delay();
+		i2c_master_set_scl_float_and_wait_ready();
```

Comment:
Check the return code and return it

Reply:
same as the other update.

Reply:
i have made a bit change to handle the error inside i2c mode in a unified place, hence they are tracked together now. resolved as dup for now, so we have a single place for discussion.